src/fips/indicators.rs

```diff
@@ -841,6 +850,51 @@ const FIPS_CHECKS: FipsChecks = FipsChecks {
                 | CKF_UNWRAP
                 | CKF_DERIVE,
         },
+        FipsMechanism {
+            mechanism: CKM_TLS12_MASTER_KEY_DERIVE,
+            operations: CKF_DERIVE,
+            restrictions: [
+                restrict!(CKK_GENERIC_SECRET, step!(48 * 8)),
+                restrict!(),
+            ],
+            genflags: CKF_SIGN
+                | CKF_VERIFY
+                | CKF_ENCRYPT
+                | CKF_DECRYPT
+                | CKF_WRAP
+                | CKF_UNWRAP
```

Comment:
Are we sure that keys derived from the TLS master key can be used to wrap/unwrap or further derive other keys ?
Mac and Encryption are obviously ok.

Reply:
The PKCS#11 specs are vague and probably written by people who understand the TLS 1.2. The notes have the following wording (hope it is not for 1.0 and 1.1):

> The mechanism also contributes the CKA_ALLOWED_MECHANISMS attribute consisting only of CKM_TLS12_KEY_AND_MAC_DERIVE, CKM_TLS12_KEY_SAFE_DERIVE, CKM_TLS12_KDF and CKM_TLS12_MAC.

From there I read that the master key derived keys can be used for all sign/verify/encrypt/decrypt/derive. The wrap/unwrap really does not make sense, right.